apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

```diff
@@ -4,38 +4,54 @@
 import logging
 import os
 import sys
-from typing import Generator, Optional, Tuple
+from typing import Any, Dict, Generator, Optional, Tuple
 
 import click
+import pystac
 from datacube import Datacube
 from datacube.index.hl import Doc2Dataset
-from odc.apps.dc_tools.utils import (
-    allow_unsafe,
-    index_update_dataset,
-    limit,
-    update_if_exists,
-)
+from odc.apps.dc_tools.utils import (allow_unsafe, index_update_dataset, limit,
+                                     update_if_exists)
 from odc.index.stac import stac_transform, stac_transform_absolute
-from satsearch import Search
+from pystac_client import Client
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s: %(levelname)s: %(message)s', datefmt='%m/%d/%Y %I:%M:%S')
+
+
+def _parse_options(options: Optional[str]) -> Dict[str, Any]:
+    parsed_options = {}
 
+    if options is not None:
+        for option in options.split("#"):
+            try:
+                key, value = option.split("=")
+                parsed_options[key] = value
+            except Exception as e:
+                logging.warning(f"Couldn't parse option {option}, format is key=value, exception was {e}")
 
-def guess_location(metadata: dict) -> Tuple[str, bool]:
+    return parsed_options
+
+
+def _guess_location(item: pystac.Item) -> Tuple[str, bool]:
     self_link = None
     asset_link = None
     relative = True
 
-    for link in metadata.get("links"):
-        rel = link.get("rel")
-        if rel and rel == "self":
-            self_link = link.get("href")
+    for link in item.links:
+        if link.rel == "self":
+            self_link = link.target
 
-    if metadata.get("assets"):
-        for asset in metadata["assets"].values():
-            if asset.get("type") in [
-                "image/tiff; application=geotiff; profile=cloud-optimized",
-                "image/tiff; application=geotiff",
-            ]:
-                asset_link = os.path.dirname(asset["href"])
+        # Override self with canonical
+        if link.rel == "canonical":
+            self_link = link.target
+
+    for name, asset in item.assets.items():
+        if asset.media_type in [
```

Comment:
This won't get USGS new standard which is **image/vnd.stac.geotiff; cloud-optimized=true"**

isn't it better to change to 
`if 'geotiff' in asset.media_type`

Reply:
It's better to use the `role` of data, really...

But USGS make their QA bands "metadata" role, so that doesn't help.

I'll do what you suggest, I think it is more flexible :-)